niwidgets/ciftitools.py

```diff
@@ -0,0 +1,268 @@
+from __future__ import print_function
+import os, sys, time
+import numpy as np
+import nibabel as nb
+
+#class CiftiTools:
+#    '''
+#    Tools to manipulate cifti files and prepare them for display.
+#    '''
+#
+#    def __init__(self, filename):
+#        if not os.path.exists(filename):
+#            raise ValueError('File {}does not exist, please provide a valid path.'.format(filename))
+#        self.filename=filename
+
+def cread(infile):
+    '''
+    Loads a cifti file and stores as five objects: data array, cifti header, nifti header, extra metadata, filemap
+    Example call:
+        cd,ch,cn,cx,cf=cread(path_to_cifti)
+    
+    Parameters
+    ----------
+    infile: str, Cifti2Image
+        string containing path to cifti file or an already loaded cifti object
+    '''
+    if isinstance(infile,nb.cifti2.cifti2.Cifti2Image):
+        try:
+            cin=infile
+        except:
+            raise ValueError('Please provide a valid cifti file.')
+    elif isinstance(infile,str):
+        if not os.path.exists(infile):
+            raise ValueError('File does not exist, please provide a valid path to a cifti file.')
+        try:
+            cin=nb.load(infile) # NB causes apparently benign warning as of NiBabel version 2.2.0dev
+        except:
+            raise ValueError('Please provide a valid cifti file.')
+    cdata=np.asarray(cin.get_data()).copy()
+    chead=cin.header.copy()
+    cnhdr=cin.nifti_header.copy()
+    cxtra=cin.extra.copy()
+    cfmap=cin.file_map.copy()
+    return cdata,chead,cnhdr,cxtra,cfmap
+
+def csave(filename,cdata,chead,cnhdr=None,cxtra=None,cfmap=None,returnImg=False):
+    '''
+    Saves a cifti image from a collection of cifti substructures to file.
+    
+    Parameters
+    ----------
+    filename: str
+        string containing output file path
+    cdata: numpy ndarray
+        array containing data to be saved, can be 1D (single timepoint) or 2D (timeseries)
+    chead: Cifti2Header
+        cifti header data, generally extracted from a previously read cifti file
+    cnhdr: Nifti2Header (optional)
+        nifti header data, also generally extracted from a previously read cifti file
+    cxtra: dict (optional)
+        extra cifti metadata, generally empty
+    cfmap: dict (optional)
+        file map, generally contains the string 'image'
+    returnImg: bool
+        optionally return the compiled cifti object
+    '''
+    cout=nb.Cifti2Image(dataobj=cdata,header=chead,nifti_header=cnhdr, extra=cxtra, file_map=cfmap)
+    cout.to_filename(filename)
+    if returnImg is True:
+        return cout
+
+def clthresh(data,thresh,setTo=0):
```

Comment:
the `setTo` keyword argument looks different in different places (eg. here and at line 90) - I'd be in favour of something that follows snake_case, so `set_to`.